### DISCUSSION TITLE
Zero Config / Example Config

### DISCUSSION THREAD

**Discussion by sauterl** [ppanopticon/vitrivr-ng#139] [Ideas]
In order to simplify various deployments and so that vitrivr-ng works as good as possible out of the box, I suggest to introduce a new zero config policy.

The current [config.json](https://github.com/vitrivr/vitrivr-ng/blob/master/src/config.json) will be removed and specifically removed from version control. Instead, we provide one or two `example.config.json` with reasonable settings.
The [config model](https://github.com/vitrivr/vitrivr-ng/blob/master/src/app/shared/model/config/config.model.ts) should provide sensible and reasonable defaults for an out of the box [Cineast](https://github.com/vitrivr/cineast).

Another advantage of this approach would be to not have to worry about the config during development.

**Comment by lucaro** [comment]
I like the idea in principle. I'm just not sure what a reasonable default would be. Should this config only include the minimum functionality and thereby ensure that everything accessible presumably also generates results, or should it showcase all the querying options?

**Reply by silvanheller** [reply]
The current config of Cineast includes reasonable features for video & images and audio fingerprinting, and vitrivr-ng includes things which are interesting for video & audio. I'm in favor of minimum functionality which still showcases interesting work.
